lib/rack/request.rb

```diff
@@ -645,6 +651,8 @@ def parse_http_accept_header(header)
           end
           [attribute, quality]
         end
+
+        parts.compact
```

Comment:
`parts.compact!; parts` to save another array allocation.

Reply:
Sorry, I missed this but intended to do that. Thanks for calling it out.